src/runner.js

```diff
@@ -132,6 +132,13 @@ assign(Runner.prototype, {
     this.builder.emit('query', assign({ __knexUid, __knexTxId }, obj));
 
     const runner = this;
+
+    if (global.print) {
```

Comment:
this looks quite intrusive and relies on global state. Can't this be rewritten to be some param for .migrate() operation?

Reply:
By the looks of it, currently client.js instantiates new runner and then passes builder to it. Cleaner way to implement this functionality would be to have a "dry-run" runner that prints out queries instead of executing them; hence this could be a parameter in client configuration based on which client would resolve the kind of runner it needs to instantiate.